internal/announce/service.go

```diff
@@ -86,6 +87,11 @@ func (s *service) Process(release *domain.Release) {
 
 		var rejections []string
 
+		// sleep for the delay period specified in the filter before running actions
+		delay := time.Duration(release.Filter.Delay)
+		log.Info().Msgf("Delaying processing of '%v' (%v) for %v by %d seconds as specified in the filter", release.TorrentName, release.Filter.Name, release.Indexer, delay)
+		time.Sleep(delay * time.Second)
+
```

Comment:
The default value of Delay will be 0, so a small enhancement could be to wrap this block in an `if ...Delay > 0`, and change log level to `Debug` instead of `Info`.

Reply:
I made it info because I thought it might be confusing to see the `Matched` log and then no indicator of any action taking place, but it looks like the action logs are debug anyway, so that makes sense.